Add ability to get random word

fzembow opened the pull request:
Implements #5 

Taking advantage of the trie structure to quickly get random words.

Also relaxed the constraint that prefixes be non zero-length strings. This way, `getWords` and `getPrefix` can reuse the same backing implementation.

lyndseybrowning commented:
This looks good but I'd like to review properly later :)

lyndseybrowning commented on src/recurseRandomWord.js:
```diff
@@ -0,0 +1,12 @@
+
+export default function recurseRandomWord(node, prefix) {
+  const word = prefix;
+
```
Remove empty line on line 5

lyndseybrowning commented on src/recurseRandomWord.js:
```diff
@@ -0,0 +1,12 @@
+
+  const branches = Object.keys(node);
+  const branch = branches[Math.floor(Math.random() * branches.length)];
+  if(branch === config.END_WORD) {
```
Add line break above if statement

lyndseybrowning commented on src/index.js:
```diff
@@ -96,11 +97,34 @@ export default function(input) {
       }
 
-      const { prefixNode } = checkPrefix(trie, strPrefix);
+      let prefixNode;
```
How about making prefixNode a ternary. I try to avoid using let wherever I can. Thoughts?

lyndseybrowning commented:
@fzembow Are you able to make the requested changes? I'm happy to make them if you are unable to do so as I'd like to get these changes merged in :)

fzembow commented:
@lyndseybrowning my apologies, in the intervening period I left my job and moved apartments so just getting back into things now :)

Addressed your changes, let me know if you see any other issues.

Cheers,
f

lyndseybrowning commented:
Lovely, looks good. Thanks 👍
